upstairs/src/lib.rs

```diff
@@ -3706,6 +3741,29 @@ async fn process_new_io(
     }
 }
 
+#[derive(Debug, Serialize)]
+pub struct Arg {
+    up_count: u32,
+    ds_count: u32,
+    ds_state: Vec<DsState>,
+}
+
+/**
+ * Update the counters used by dtrace probes.
+ * This one method will update the fields of the
+ * up_status counter.
+ */
+fn stat_update(up: &Arc<Upstairs>, msg: &str) {
```

Comment:
Not sure if it matters at this point, but you _could_ further reduce the likelihood of a disabled-probe effect by either inlining this function, or calling the macro directly in place of calls to this function. Totally up to you, just a thought.

Reply:
That is a good idea.  I'm going to be adding a bunch more stats, and I think I'll be
throwing this function out and sprinkling things elsewhere.  I wanted to make a first
pass at this to make sure I had the right idea how to do it.